syndeo-template/src/main/java/com/google/cloud/syndeo/SyndeoTemplate.java

```diff
@@ -173,29 +178,60 @@ public static ConfiguredSchemaTransform buildFromJsonConfig(JsonNode transformCo
         transformProvider.getClass().getProtectionDomain().getCodeSource().getLocation().getPath());
     List<Object> configurationParameters =
         transformProvider.configurationSchema().getFields().stream()
-            .map(field -> field.getName())
-            .map(fieldName -> params.has(fieldName) ? params.get(fieldName) : null)
+            .map(field -> KV.of(field, field.getName()))
             .map(
-                fieldNode ->
-                    fieldNode == null
-                        ? null
-                        : fieldNode.isBoolean()
-                            ? fieldNode.asBoolean()
-                            : fieldNode.isFloatingPointNumber()
-                                ? fieldNode.asDouble()
-                                : fieldNode.isNumber()
-                                    ? fieldNode.asLong()
-                                    : !fieldNode.isContainerNode()
-                                        ? fieldNode.asText()
-                                        : fieldNode.isArray()
-                                            ? new ObjectMapper().convertValue(fieldNode, List.class)
-                                            : new ObjectMapper().convertValue(fieldNode, Map.class))
+                fieldPair ->
+                    params.has(fieldPair.getValue())
+                        ? KV.of(fieldPair.getKey(), params.get(fieldPair.getValue()))
+                        : null)
+            .map(fieldPair -> parseJsonNode(fieldPair))
             .collect(Collectors.toList());
+
     return new ProviderUtil.TransformSpec(
             transformConfig.get("urn").asText(), configurationParameters)
         .toProto();
   }
 
+  private static Object parseJsonNode(KV<Field, JsonNode> fieldPair) {
+    if (fieldPair == null) return null;
+
+    Field field = fieldPair.getKey();
+    JsonNode fieldNode = fieldPair.getValue();
+
+    Object parsedField =
+        fieldNode == null
+            ? null
+            : fieldNode.isBoolean()
+                ? fieldNode.asBoolean()
+                : fieldNode.isFloatingPointNumber()
+                    ? fieldNode.asDouble()
+                    : fieldNode.isNumber()
+                        ? fieldNode.asLong()
+                        : !fieldNode.isContainerNode()
+                            ? fieldNode.asText()
+                            : fieldNode.isArray()
+                                ? new ObjectMapper().convertValue(fieldNode, List.class)
+                                : field.getType().getTypeName() == TypeName.MAP
+                                    ? new ObjectMapper().convertValue(fieldNode, Map.class)
+                                    : field.getType().getTypeName() == TypeName.ROW
+                                        ? convertJsonNodeToRow(fieldPair)
+                                        : new ObjectMapper().convertValue(fieldNode, Object.class);
```

Comment:
I think if () and else if () blocks would look much cleaner and less error prone here